silica/verilog.py

```diff
@@ -99,8 +101,68 @@ def compile_state_by_path(state, index, _tab, one_state, width_table):
         verilog_source += f"\n{_tab}end"
     return verilog_source, temp_var_source
 
+def get_by_name(module, name):
+    return module.get_ports().get(name, module.get_vars().get(name))
+
+# TODO: these translate functions should be merged
+# TODO: the only reason translate takes in a module is to do the name lookup. should figure out how to not need to pass in module.
```

Comment:
Perhaps wrap these functions in a class which is constructed with module, so all these methods can refer to the instance attribute `self.module`?

Reply:
Yeah, I have a few ideas of how to refactor this and what other helper functions would be useful. I'm definitely making a context class with the module where all of these will be members.